tailcfg/tailcfg.go

```diff
@@ -1698,10 +1699,17 @@ var FilterAllowAll = []FilterRule{
 	},
 }
 
+// DNSResolver embeds dnstype.Resolver and stores
+// additional configuration.
+type DNSResolver struct {
+	dnstype.Resolver `json:",omitempty"`
+	UseWithExitNode  bool `json:",omitempty"`
```

Comment:
Needs docstring, one suggestion:
```
    // UseWithExitNode designates that this resolver should continue to be used when an
    // exit node is in use. Normally, DNS resolution is delegated to the exit node but
    // there are situations where it is preferable to still use a Split DNS server and/or
    // global DNS server instead of the exit node. 
```